base/gmp.jl

```diff
@@ -458,7 +475,7 @@ powermod(x::Integer, p::Integer, m::BigInt) = powermod(big(x), big(p), m)
 
 function gcdx(a::BigInt, b::BigInt)
     if b == 0 # shortcut this to ensure consistent results with gcdx(a,b)
-        return a < 0 ? (-a,-one(BigInt),zero(BigInt)) : (a,one(BigInt),zero(BigInt))
+        return a < 0 ? (-a,-ONE,ZERO) : (a,ONE,ZERO)
```

Comment:
I didn't say the issue I linked to was supporting this change, I was merely referencing it as a relevant discussion for the problem you pointed at. `BigInt` is considered to be conceptually (if not technically) an immutable type, and it's possible that a future version of Julia gets `BigInt` implemented as `immutable/struct` (cf. e.g. https://github.com/JuliaLang/julia/pull/17015/files#diff-c39f3bbf0cb1c2381243b42647d9ff1dR43), which would break the mutating code you posted above. I don't have a strong feeling about what should be the resolution here, and will gladly modify the code according to the consensus if any is reached.